src/main/java/teammates/ui/webapi/action/GetSessionResultsAsCsvAction.java

```diff
@@ -0,0 +1,84 @@
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.EntityNotFoundException;
+import teammates.common.exception.ExceedingRangeException;
+import teammates.common.exception.InvalidHttpParameterException;
+import teammates.common.exception.RequestExceedingRangeException;
+import teammates.common.util.Const;
+
+/**
+ * Gets feedback session results in csv.
+ */
+public class GetSessionResultsAsCsvAction extends Action {
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.PUBLIC;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String feedbackSessionName = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
+
+        FeedbackSessionAttributes fs = logic.getFeedbackSession(feedbackSessionName, courseId);
+
+        if (fs == null) {
+            throw new EntityNotFoundException(new EntityDoesNotExistException("Feedback session is not found"));
+        }
+
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+        gateKeeper.verifyAccessible(instructor, fs);
+    }
+
+    @Override
+    public ActionResult execute() {
+        String section = getRequestParamValue(Const.ParamsNames.SECTION_NAME);
+        String sectionDetailValue = getRequestParamValue(Const.ParamsNames.SECTION_NAME_DETAIL);
+        SectionDetail sectionDetail = SectionDetail.NOT_APPLICABLE;
+        if (section != null && sectionDetailValue != null && !sectionDetailValue.isEmpty()) {
+            if (SectionDetail.containsSectionDetail(sectionDetailValue) == null) {
+                throw new InvalidHttpParameterException("Section detail is invalid.");
+            }
+            sectionDetail = SectionDetail.valueOf(sectionDetailValue);
+        }
+
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String feedbackSessionName = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
+        String questionId = getRequestParamValue(Const.ParamsNames.FEEDBACK_QUESTION_ID);
+        boolean isMissingResponsesShown = getBooleanRequestParamValue(
+                Const.ParamsNames.FEEDBACK_RESULTS_INDICATE_MISSING_RESPONSES);
+        boolean isStatsShown = getBooleanRequestParamValue(Const.ParamsNames.FEEDBACK_RESULTS_SHOWSTATS);
+        String simulateExcessDataForTesting = getRequestParamValue("simulateExcessDataForTesting");
+
+        String fileContent = "";
+
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+
+        try {
+            if ("true".equals(simulateExcessDataForTesting)) {
+                throw new ExceedingRangeException("This session has more responses than that can be downloaded at one go.");
+            }
+
+            if (section == null || "All".equals(section)) {
```

Comment:
I feel `"All".equals(section)` is a little hacky and hardcoded - any way to make it cleaner?

Reply:
This is ported from the `InstructorFeedbackResultsDownloadAction` class. I agree that this is a hardcoded value. Maybe we can remove it since `section == null` already implies all sections will be downloaded.